src/NetworkDynamics.jl

```diff
@@ -263,6 +268,99 @@ function network_dynamics(vertices!::Array{VertexFunction}, edges!::Array{EdgeFu
     nothing
 end
 
+"""
+If only a sinlge Function is given, not an Array of EdgeFunctions.
+"""
+function prepare_edges(edge::EdgeFunction, g::SimpleGraph)
+    if edge.coupling == :directed
+        @error "Coupling type not available for undirected Graphs"
+    elseif edge.coupling ∈ (:symmetric, :antisymmetric, :unspecified)
+        return reconstruct_edge(edge)
+    end
+end
+
+""" prepare_edges(edges, g::SimpleGraph)
+
+
+"""
+
+function prepare_edges(edges, g::SimpleGraph)
+    new_edges = similar(edges)
+    for (i, edge) in enumerate(edges)
+        if edge.coupling == :directed
+            @error "Coupling type of edge $i not available for undirected Graphs"
+        elseif edge.coupling ∈ (:symmetric, :antisymmetric, :unspecified)
+            new_edges[i] = reconstruct_edge(edge)
+        else
+            new_edges[i] = edges[i]
+        end
+    end
+    return new_edges
+end
+
+"""
+"""
+function prepare_edges(edges, g::SimpleDiGraph)
+    for (i, edge) in enumerate(edges)
+        if edge.coupling ∈ (:symmetric, :antisymmetric, :undirected)
+            @error "Coupling type of edge $i not available for directed Graphs"
+        #This does not work because StaticEdge is immutable. But maybe its not necessary.
+        #elseif edge.coupling == :unspecified
+        #    edge.coupling = :directed
+        end
+    end
+    edges
+end
+
+# prepare_edge should be simplified!
+function prepare_edges(edge::EdgeFunction, g::SimpleDiGraph)
+    if edge.coupling ∈ (:symmetric, :antisymmetric, :undirected)
+        @error "Coupling type of EdgeFunction not available for directed Graphs"
+    end
+    edge
+end
+
+
+@inline function reconstruct_edge(edge::StaticEdge)
+    # Unfortunately leads to a very messy signature
+    let dim = edge.dim, orig_f = edge.f!
+      if edge.coupling == :unspecified
+          # This might cause unexpected behaviour if source and destination vertex don't have
+          # the same internal arguments
+          # Make sure to explicitly define the edge is :undirected in that case.
+          f! = @inline (e, v_s, v_d, p, t) -> begin
+              @inbounds orig_f(view(e,1:dim), v_s, v_d, p, t)
+              @inbounds orig_f(view(e,dim+1:2dim), v_d, v_s, p, t)
+          end
+      elseif edge.coupling == :antisymmetric
+          f! = @inline (e, v_s, v_d, p, t) -> begin
+              @inbounds orig_f(view(e,1:dim), v_s, v_d, p, t)
+              @inbounds view(e,dim+1:2dim) .= -view(e,1:dim)
```

Comment:
```suggestion
              @inbounds view(e,dim+1:2dim) .= -1.0 .* view(e,1:dim)
```
gets rid of the last allocation.
`Float64 * AbstractArray` creates a new array (allocating).
Might be faster to loop through the dimensions though, havn't testet it.

Reply:
wow :D
wouldn't have thought its only such small bugs. Thanks for finding those!